File: 0035-search-insert-position/0035-search-insert-position.go
func searchInsert(nums []int, target int) int {
    n:=len(nums)
    low:=0
    high:=n-1
    res:=n

    for low<=high{
        guess:=(low+high)/2

        if nums[guess]>=target{
            res=guess
            high=guess-1
        }else{
            low=guess+1
        }
    }
    return res
}

File: 0035-search-insert-position/0035-search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int n=nums.length;
        int low=0;
        int high=n-1;
        int res=n;

        while(low<=high){
            int guess=(low+high)/2;

            if(nums[guess]>=target){
                res=guess;
                high=guess-1;
            }
            else{
                low=guess+1;
            }
        }
        return res;
    }
}